### ISSUE TITLE
Default ContinuousTimeMSM convergence tolerance?

### ISSUE THREAD

**Issue by rmcgibbo** [msmbuilder/msmbuilder#451]
Using the following script with `ContinuousTimeMSM` and the `MullerPotential`, it's clear that the estimate of the longest timescale depends significantly on the BFGS convergence tolerance. Obviously the runtime also depends pretty strongly on the convergence tolerance, so it's not clear what the right value for the default is.

Currently the default `ftol` is `1e-6`. Perhaps it should be moved to `1e-8`.

Results:
```
ftol = 1e-06: 0.947 seconds
[ 346.06811842   11.55206465    3.8997996     2.61397625]
ftol = 1e-07: 1.523 seconds
[ 402.89423991   11.88789696    4.0831871     2.72734776]
ftol = 1e-08: 3.026 seconds
[ 460.86207884   12.05381497    4.15459442    2.73196869]
ftol = 1e-09: 5.157 seconds
[ 469.02854103   12.06107094    4.1735615     2.73451371]
ftol = 1e-10: 9.794 seconds
[ 474.79438021   12.07621162    4.17715944    2.73720923]
```

Script:
```
import warnings
from mdtraj.utils import timing
from msmbuilder.msm import ContinuousTimeMSM
from msmbuilder.example_datasets import MullerPotential
from msmbuilder.cluster import NDGrid
from sklearn.pipeline import Pipeline

X = MullerPotential().get()['trajectories']

warnings.simplefilter('ignore')

pipeline = Pipeline([
    ('cluster', NDGrid(n_bins_per_feature=10)),   # 100 states
    ('msm', ContinuousTimeMSM(n_timescales=4))
])

for ftol in [1e-6, 1e-7, 1e-8, 1e-9, 1e-10]:
    pipeline.set_params(msm__ftol=ftol)

    with timing('ftol = %g' % ftol):
        pipeline.fit(X)
    print(pipeline.named_steps['msm'].timescales_)
```

**Comment by kyleabeauchamp** [comment]
IMHO defaults should be conservative: 10^-10 for now?

We can always tune the optimizer later, lots of room to experiment later

**Comment by rmcgibbo** [comment]
Well that experiment didn't try using a tolerance even lower. Presumably you could go all the way down to ~1e-16 for machine epsilon.

**Comment by rmcgibbo** [comment]
There is a point of diminishing marginal returns.

**Comment by kyleabeauchamp** [comment]
Near the minimum, you might get faster O(x^2) convergence as you see Newton-like behavior though?

**Comment by rmcgibbo** [comment]
Going to 1e-13, i'm not sure I see an evidence of that.
```
ftol = 1e-06: 1.327 seconds
[ 475.55376155   12.80257627    4.9400912     3.86664567]
ftol = 1e-07: 1.819 seconds
[ 552.74025419   12.86299248    6.37798983    3.93131254]
ftol = 1e-08: 2.895 seconds
[ 604.42630442   12.96460165    5.89946752    3.97616693]
ftol = 1e-09: 6.088 seconds
[ 628.64786444   12.97734266    6.58756727    4.02590946]
ftol = 1e-10: 11.761 seconds
[ 633.79699182   12.98719812    6.62382215    4.03763376]
ftol = 1e-11: 31.207 seconds
[ 638.98788579   12.99352569    6.71827186    4.04821537]
ftol = 1e-12: 39.335 seconds
[ 639.34952105   12.99374239    6.70430519    4.04935552]
ftol = 1e-13: 38.983 seconds
[ 639.34952105   12.99374239    6.70430519    4.04935552]
```

**Comment by kyleabeauchamp** [comment]
Well, I looked through the code and didn't see any obvious pathways to improvement.  I imagine something like BFGS-SGD might work, but that's a research-level project.

Regarding default tolerance, my gut feeling here is 1E-12.  I think users envision MSMBuilder as a "turnkey" solution and will only adjust hyperparameters that are directly discussed in the primary literature (e.g. lagtime).  CYA

**Comment by rmcgibbo** [comment]
I changed the default to 1e-10. We can continue to play around with this in the future. It's also possible to use a termination criteria on the gradient norm instead of the objective function (or in addition I think), which is another knob.
